### ISSUE TITLE
Libtorch Don't Suppory Multi-threaded

### ISSUE THREAD

**Issue by a819411321** [pytorch/pytorch#117938]
### 🐛 Describe the bug

my code as following:
-------------------------------------
Init(0, weigth_path);
Init(1, weigth_path);
std::thread t0 = std::thread(test0);//test0 is inference such as auto outputs = model.forward(inputs).toTuple(); 
std::thread t1 = std::thread(test1);//test1 is inference
t0.detach();
t1.detach();
-------------------------------------
In only thread t0 inference time is 10ms and in only thread t1 inference time is also 10ms.
but when thread t0 and thread t1 both inference , time is 20ms. It looks like there's one thread waiting for another

### Versions

libtorch 2.0.1
